src/screens/RegisterEnsSheet.js

```diff
@@ -35,18 +37,26 @@ export default function RegisterEnsSheet() {
   const [searchQuery, setSearchQuery] = useState('');
   const debouncedSearchQuery = useDebounceString(searchQuery);
 
+  const { getAvailable, getRentPrice, getNameExpires } = useENS();
   const { data: registration, status } = useQuery(
     debouncedSearchQuery.length > 2 && ['registration', debouncedSearchQuery],
     async (_, searchQuery) => {
       const fastFormatter = timestamp => {
         const date = new Date(Number(timestamp) * 1000);
         return `${date.toDateString()}`;
       };
-      const registration = await fetchRegistration(searchQuery + '.eth');
+      const isAvailable = await getAvailable(searchQuery);
+      // dummy default to 1 year
+      const rentPrice = await getRentPrice(searchQuery, 31536000);
+      const exipryDate = await getNameExpires(searchQuery);
+      const { registrationDate } = await fetchRegistration(
+        searchQuery + '.eth'
+      );
       return {
-        expiryDate: fastFormatter(registration.expiryDate),
-        isRegistered: registration.isRegistered,
-        registrationDate: fastFormatter(registration.registrationDate),
+        expiryDate: fastFormatter(exipryDate),
+        isRegistered: !isAvailable,
+        registrationDate: fastFormatter(registrationDate),
+        rentPrice: fromWei(rentPrice.toString()),
```

Comment:
If we refactor the current `useENS` implementation into their helpers (previous suggestion), then we could even repurpose `useENS` to implement this `useQuery` hook (lines 41 to 62), and the usage could be:

```tsx
const { data, status } = useENS({ name: debouncedSearchQuery })
```

Reply:
I wonder if `useENS` is the right naming, people may think that it is a hook to get ENS details of a name. Maybe we should call this `useENSRegistration`?

Reply:
i like this idea 💯